src/features/session_replay/shared/recorder.js

```diff
@@ -74,8 +83,39 @@ export class Recorder {
     }
   }
 
+  /**
+   * audit - Checks if the event node payload is missing certain attributes
+   * will forward on to the "store" method if nothing needs async fixing
+   * @param {*} event - An RRWEB event node
+   * @param {*} isCheckout - Flag indicating if the payload was triggered as a checkout
+   */
+  audit (event, isCheckout) {
+    /** only run the audit if inline_stylesheets is configured as on (default behavior) */
+    if (this.shouldInlineStylesheets) {
+      this.currentBufferTarget.inlinedAllStylesheets = false
+      return this.store(event, isCheckout)
+    }
+    /** An count of stylesheet objects that were blocked from accessing contents via JS */
+    const incompletes = stylesheetEvaluator.evaluate()
+    /** Only stop ignoring data if already ignoring and a new valid snapshap is taking place (0 incompletes and we get a meta node for the snap) */
+    if (!incompletes && this.#fixing && event.type === RRWEB_EVENT_TYPES.Meta) this.#fixing = false
+    if (incompletes) {
+      handle(SUPPORTABILITY_METRIC_CHANNEL, ['SessionReplay/Payload/Missing-Inline-Css', incompletes], undefined, FEATURE_NAMES.metrics, this.parent.ee)
+      /** wait for the evaluator to download/replace the incompletes' src code and then take a new snap */
+      stylesheetEvaluator.fix().then((failedToFix) => {
+        if (failedToFix) this.currentBufferTarget.inlinedAllStylesheets = false
+        this.takeFullSnapshot()
+      })
+      /** Only start ignoring data if got a faulty snapshot */
+      if (event.type === RRWEB_EVENT_TYPES.FullSnapshot || event.type === RRWEB_EVENT_TYPES.Meta) this.#fixing = true
```

Comment:
Is this supposed to wait on L107 or referring to that snapshot?

Reply:
it will trigger before the async .then, so all that really matters is that `#fixing === true` is in control of whether the module stores the data.  Once the async new snapshot occurs, it will unset the fixing (#fixing === false) and store will be successful